Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorHelpers.cpp

```diff
@@ -382,11 +382,28 @@ namespace AzToolsFramework
 
     bool EditorHelpers::IsSelectableInViewport(const AZ::EntityId entityId) const
     {
-        return IsSelectableAccordingToFocusMode(entityId) && IsSelectableAccordingToContainerEntities(entityId);
+        auto entityCacheId = m_entityDataCache->GetVisibleEntityIndexFromId(entityId);
+        if (entityCacheId.has_value())
+        {
+            return IsSelectableInViewport(entityCacheId.value());
+        }
+
+        return false;
+    }
+
+    bool EditorHelpers::IsSelectableInViewport(size_t entityCacheId) const
```

Comment:
nit: Could you please update the parameter name to be `entityCacheIndex`

Reply:
Fixed everywhere!

Reply:
Thank you! 😄